crates/ethereum-forks/src/forkcondition.rs

```diff
@@ -108,3 +116,182 @@ impl ForkCondition {
+        );
+        // Test with earlier timestamp
+        assert!(
+            !fork_condition.transitions_at_timestamp(123, 122),
+            "The condition should not transition if the parent timestamp is earlier"
+        );
+    }
+
+    #[test]
+    fn test_active_at_head() {
+        let head = Head {
+            hash: Default::default(),
+            number: 10,
+            timestamp: 12345,
+            total_difficulty: U256::from(1000),
+            difficulty: U256::from(100),
+        };
+
+        // Test if the condition activates based on block number
+        let fork_condition = ForkCondition::Block(10);
+        assert!(
+            fork_condition.active_at_head(&head),
+            "The condition should be active at the given head block number"
+        );
+        let fork_condition = ForkCondition::Block(11);
+        assert!(
+            !fork_condition.active_at_head(&head),
+            "The condition should not be active at the given head block number"
+        );
+
+        // Test if the condition activates based on timestamp
+        let fork_condition = ForkCondition::Timestamp(12345);
+        assert!(
+            fork_condition.active_at_head(&head),
+            "The condition should be active at the given head timestamp"
+        );
+        let fork_condition = ForkCondition::Timestamp(12346);
+        assert!(
+            !fork_condition.active_at_head(&head),
+            "The condition should not be active at the given head timestamp"
+        );
+
+        // Test if the condition activates based on total difficulty and block number
+        let fork_condition =
+            ForkCondition::TTD { fork_block: Some(9), total_difficulty: U256::from(900) };
+        assert!(
+            fork_condition.active_at_head(&head),
+            "The condition should be active at the given head total difficulty"
+        );
+        let fork_condition =
+            ForkCondition::TTD { fork_block: None, total_difficulty: U256::from(900) };
+        assert!(
+            fork_condition.active_at_head(&head),
+            "The condition should be active at the given head total difficulty as the block number is unknown"
+        );
+        let fork_condition =
+            ForkCondition::TTD { fork_block: Some(11), total_difficulty: U256::from(900) };
+        assert!(
+            !fork_condition.active_at_head(&head),
+            "The condition should not be active as the block is higher than head"
+        );
+
+        // Even if the block number is correct compare to head, the condition should not activate if
+        // the total difficulty is too high
+        let fork_condition =
+            ForkCondition::TTD { fork_block: Some(10), total_difficulty: U256::from(9000) };
+        assert!(
+            !fork_condition.active_at_head(&head),
+            "The condition should not be active as the total difficulty is higher than head"
+        );
+    }
```

Comment:
this condition should be active because the fork block is higher

Reply:
Fixed